lib/rspec/core/formatters/documentation_formatter.rb

```diff
@@ -22,7 +22,7 @@ def example_group_started(notification)
         end
 
         def example_group_finished(_notification)
-          @group_level -= 1
+          @group_level =  @group_level > 0 ? @group_level - 1 : @group_level
```

Comment:
This is a private api, it doesn't need to return anything so `@group_level -= 1 if @group_level > 0` is preferred.